app/sidekiq/user_action_events_cleanup_job.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class UserActionEventsCleanupJob
+  include Sidekiq::Job
+
+  sidekiq_options unique_for: 30.minutes, retry: false
+
+  EXPIRATION_TIME = 1.year
+
+  def perform
+    events = UserActionEvent.where('created_at < ?', EXPIRATION_TIME.ago)
```

Comment:
can you use a time range here, just to leverage ActiveRecord a bit more directly?

`UserActionEvent.where(created_at: ...EXPIRATION_TIME.ago)`